rs/nns/test_utils/src/state_test_helpers.rs

```diff
@@ -361,7 +361,7 @@ pub fn get_canister_status(
 
 pub fn get_root_canister_status(machine: &StateMachine) -> Result<CanisterStatusResultV2, String> {
     machine
-        .canister_status_as(PrincipalId::from(ROOT_CANISTER_ID), ROOT_CANISTER_ID)
+        .canister_status_as(PrincipalId::from(LIFELINE_CANISTER_ID), ROOT_CANISTER_ID)
```

Comment:
Can root not request its own canister status?

Reply:
It can't do so via an _ingress_ message since in production a canister would request its own status via an _inter-canister_ call. Now this PR adds the ingress filtering as performed in production which prevents calls to `canister_status` from non-controllers.

Reply:
I'm wondering if state_machine could eventually have a feature that adds the ability to send inter-canister calls directly?  That was a great feature that allowed us to avoid using the universal canister when doing the testing.

Reply:
> I'm wondering if state_machine could eventually have a feature that adds the ability to send inter-canister calls directly? That was a great feature that allowed us to avoid using the universal canister when doing the testing.

The only easy option I see is to have the StateMachine test API deploy a universal canister behind the scenes and proxy the call through it. But then the caller of the proxied call would be that universal canister which might be an issue in the case of mgmt canister calls.

A more involved approach might be to enqueue an actual inter-canister call as if it was submitted during an actual canister message execution, specifying -1 as the callbacks and marking the parent call context as deleted. But this sounds pretty ad-hoc and I'm not sure if this wouldn't violate other assumptions in the system.

Reply:
Hm - I'm not sure if this is actually a good change to make to our testing tools, if we are going to lose the ability to just make simple calls as other canisters.  It may make StateMachine more faithful to the replica, but is that faithfulness important at the level of testing canister APIs?

Reply:
> if we are going to lose the ability to just make simple calls as other canisters

you're not losing that ability completely, only in cases where the ingress filter would reject such a message

Reply:
And, e.g., ingress messages to the `canister_info` endpoint are already rejected even without this PR.

Reply:
> you're not losing that ability completely, only in cases where the ingress filter would reject such a message

Okay, then that's probably fine.  As long as we can still send a message from Canister principal, this is okay.